terraform/modules/web/README.md

```diff
@@ -0,0 +1,154 @@
+# CDAP Web Module
+
+This module creates a CloudFront distribution and origin access control intended for use with the AB2D, BCDA and DPC static websites. A sample minimal calling configuration is as follows:
+
+```
+module "platform" {
+  source    = "../modules/platform"
+  providers = { aws = aws, aws.secondary = aws.secondary }
+
+  app         = "bcda"
+  env         = "prod"
+  root_module = ""
+  service     = "bcda"
+}
+
+module web_acl {
+  source  = "../modules/firewall"
+
+  app           = module.platform.app
+  content_type  = "APPLICATION_JSON"
+  env           = module.platform.env
+  name          = "samplewebacl"
+  scope         = "CLOUDFRONT"
+}
+
+module origin_bucket {
+  source  = "../modules/bucket"
+  
+  app   = module.platform.app
+  env   = module.platform.env
+  name  = "origin"
+}
+
+module logging_bucket {
+  source  = "../modules/bucket"
+  
+  app   = module.platform.app
+  env   = module.platform.env
+  name  = "logging"
+}
+
+resource "aws_acm_certificate" "cert" {
+  domain_name       = "bcda.cms.gov"
+  validation_method = "DNS"
+
+  tags = {
+    Environment = "test"
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+module "web" {
+  source = "../modules/web"
+
+  certificate     = aws_acm_certificate.cert
+  logging_bucket  = module.logging_bucket
+  origin_bucket   = module.origin_bucket
+  platform        = module.platform
+  web_acl         = module.web_acl
+  
+  viewer_request_function_list = [
+    {
+      code    = "test_code1"
+      comment = "test_comment1"
+      name    = "test_name1"
+      runtime = "cloudfront-js-2.0"
+    },
+    {
+      code    = "test_code2"
+      comment = "test_comment2"
+      name    = "test_name2"
+      runtime = "cloudfront-js-2.0"
+    }
+  ]
+}
```

Comment:
This is looking good! I'd still like to see the request functions standardized in the module. Looks like between https://github.com/CMSgov/ab2d/blob/main/ops/services/15-web/redirects-function.js and https://github.com/CMSgov/bcda-ops/blob/main/terraform_gf/static_site/modules/cloudfront/redirects-function.js the only real difference is the redirects. Maybe those could be passed into the module and added to a templatized redirects-function.js?